applications/FluidDynamicsApplication/python_scripts/apply_two_fluids_inlet_process.py

```diff
@@ -0,0 +1,219 @@
+                    v1 = settings["fluid_1_settings"]["direction"].GetVector()
+                    v2 = settings["fluid_2_settings"]["direction"].GetVector()
+                    if ( abs(v1[0] - v2[0]) > 1.0e-7 or abs(v1[1] - v2[1]) > 1.0e-7 or abs(v1[2] - v2[2]) > 1.0e-7 ):
+                        raise Exception("The values (vector) of the entry 'direction' differ between the two fluids. This is not admissible.")
+            else:
+                raise Exception("The values (different type) of the entry 'direction' differ between the two fluids. This is not admissible.")
+
+        self.inlet_model_part = Model[settings["model_part_name"].GetString()]
+        self.complete_model = self.inlet_model_part.GetRootModelPart()
+
+        self.neighbour_search = KratosMultiphysics.FindNodalNeighboursProcess(self.complete_model, 10, 10)
+        self.neighbour_search.Execute()
+
+        if ( self.complete_model.GetCommunicator().TotalProcesses() == 1 ):
+            # operation for serial or OpenMP
+            self.variational_distance_process = self._set_variational_distance_process_serial()
+            self.two_fluid_inlet_process = KratosFluid.TwoFluidsInletProcess( self.inlet_model_part, settings["interface_settings"], self.variational_distance_process )
+
+        elif ( self.complete_model.GetCommunicator().TotalProcesses() > 1 ):
+            # operation for MPI (based on Trilinos)
+            self.variational_distance_process = self._set_variational_distance_process_mpi()
+            self.two_fluid_inlet_process = KratosFluid.TwoFluidsInletProcess( self.inlet_model_part, settings["interface_settings"], self.variational_distance_process )
+
+
+        import assign_vector_by_direction_process
+
+        if ( self.inlet_model_part.GetSubModelPart("fluid_1_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_1 = assign_vector_by_direction_process.AssignVectorByDirectionProcess(Model, settings["fluid_1_settings"])
+
+        if ( self.inlet_model_part.GetSubModelPart("fluid_2_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_2 = assign_vector_by_direction_process.AssignVectorByDirectionProcess(Model, settings["fluid_2_settings"])
+
+
+
+    def ExecuteInitializeSolutionStep(self):
+        # Call the base process ExecuteInitializeSolutionStep()
+        if ( self.inlet_model_part.GetSubModelPart("fluid_1_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_1.ExecuteInitializeSolutionStep()
+        if ( self.inlet_model_part.GetSubModelPart("fluid_2_inlet").NumberOfNodes() > 0):
+            self.aux_process_fluid_2.ExecuteInitializeSolutionStep()
+
+        # Not sure if appropriate here...
+        # PROBLEM: Could distort the physical properties
+        # BUT: It could make sense to stabilize cases with CFL >> 1
+        self.two_fluid_inlet_process.SmoothDistanceField()
```

Comment:
As far as I understand, you are smoothing the level set close to the inlet before and after the problem resolution. Are we sure that we want to do it twice per time step?

Reply:
Of course, one time is enough - I was unsure where to put it best... After all experiences made so far with this one being commented, version with the one at the end entirely seems to work. Thnk you for the hint. I will remove it here.

Reply:
Done.